src/osx/core/hidjoystick.cpp

```diff
@@ -872,7 +872,7 @@ void* wxJoystickThread::Entry()
                 wxevent.SetEventType(wxEVT_JOY_BUTTON_UP);
             }
 
-            wxevent.SetButtonChange(nIndex+1);
+            wxevent.SetButtonChange(1 << nIndex);
```

Comment:
It might be purely theoretic, but shouldn't we check for overflow here?

Reply:
I'll add 
```wxCHECK_RET(j_evt.number < CHAR_BIT * (int) sizeof(int), (void *) -1);```